src/Microsoft.Extensions.Configuration.Azure.KeyVault/DefaultKeyVaultSecretManager.cs

```diff
@@ -14,7 +14,7 @@ public class DefaultKeyVaultSecretManager : IKeyVaultSecretManager
         /// <inheritdoc />
         public virtual string GetKey(Secret secret)
         {
-            return secret.SecretIdentifier.Name.Replace("__", ConfigurationPath.KeyDelimiter);
+            return secret.SecretIdentifier.Name.Replace("--", ConfigurationPath.KeyDelimiter);
```

Comment:
Is `:` valid in KeyVault? I think it is but I can't remember for sure. The reason that `__` is there at all is for things like environment variables on Linux and other things that don't support `:`. If `:` works then I would just tell people to use that instead of changing this.